src/main/java/org/mastodon/mamut/launcher/MastodonDndLauncher.java

```diff
@@ -0,0 +1,68 @@
+package org.mastodon.mamut.launcher;
+
+import org.mastodon.mamut.MainWindow;
+import org.mastodon.mamut.WindowManager;
+import org.mastodon.mamut.project.MamutProject;
+import org.mastodon.mamut.project.MamutProjectIO;
+
+import org.scijava.io.IOPlugin;
+import org.scijava.io.AbstractIOPlugin;
+import org.scijava.io.location.FileLocation;
+import org.scijava.io.location.Location;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import org.scijava.log.LogService;
+
+import javax.swing.WindowConstants;
+import java.io.IOException;
+import java.util.ArrayList;
+
+@Plugin(type = IOPlugin.class)
+public class MastodonDndLauncher extends AbstractIOPlugin<Object> {
+
+	@Parameter
+	private LogService logService;
+
+	@Override
+	public boolean supportsOpen(Location source) {
+		final String sourcePath = source.getURI().getPath();
+		logService.debug("MastodonDndLauncher was questioned: "+sourcePath);
+
+		if (!(source instanceof FileLocation)) return false;
+		return sourcePath.endsWith(".mastodon");
+	}
+
+	@Override
+	public Object open(Location source) throws IOException {
+		logService.debug("MastodonDndLauncher was asked to open: "+source.getURI().getPath());
+		final FileLocation fsource = source instanceof FileLocation ? (FileLocation)source : null;
+		if (fsource == null) return null; //NB: shouldn't happen... (in theory)
+
+		final String projectPath = fsource.getFile().getAbsolutePath();
+
+		System.setProperty( "apple.laf.useScreenMenuBar", "true" );
+		final WindowManager windowManager = new WindowManager( getContext() );
+
+		final MainWindow mainWindow = new MainWindow( windowManager );
+		mainWindow.setVisible( true );
+		mainWindow.setDefaultCloseOperation( WindowConstants.EXIT_ON_CLOSE );
+
+		try {
+			final MamutProject project = new MamutProjectIO().load( projectPath );
+			windowManager.getProjectManager().open( project, true );
```

Comment:
....not sure if this whole code sequence is the correct way to start mastodon, please review @tinevez